src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -81,34 +81,40 @@
     private static ProgressTracker progressTracker = null;
 
     /**
-     * Generates the authorship and commits JSON file for each repo in {@code configs} at {@code outputPath}, as
-     * well as the summary JSON file of all the repos.
+     * Generates the authorship and commits JSON file for each repo in {@code configs} at {@code outputPath}.
      *
      * @return the list of file paths that were generated.
      * @throws IOException if templateZip.zip does not exists in jar file.
      */
-    public static List<Path> generateReposReport(List<RepoConfiguration> configs, String outputPath,
-            String generationDate, Date cliSinceDate, Date untilDate,
-            boolean isSinceDateProvided, boolean isUntilDateProvided) throws IOException {
+    public static List<Path> generateReposReport(List<RepoConfiguration> configs, String outputPath)
+            throws IOException {
         prepareTemplateFile(outputPath);
-
         earliestSinceDate = null;
         progressTracker = new ProgressTracker(configs.size());
+        return cloneAndAnalyzeRepos(configs, outputPath);
+    }
 
-        List<Path> reportFoldersAndFiles = cloneAndAnalyzeRepos(configs, outputPath);
-
+    /**
+     * Generates the summary JSON file for all repos in {@code configs} at {@code outputPath}.
+     *
+     * @return the path of the generated file.
+     */
+    public static List<Path> generateSummary(List<RepoConfiguration> configs, String outputPath,
```

Comment:
I split the original `generateReposReport` method into 2 methods:

- `generateReposReport` : creates the `authorship.json` and `commits.json` file for each repo
- `generateSummary`: creates the single `summary.json` file.

I broke up `generateReposReport` to make it easier to test the functionality in this PR.
By having `generateSummary` as a separate method, the test code can pass an arbitrary value of `reportGenerationTime`.